### ISSUE TITLE
NowDock 0.5 - A Huge Release To Be...

### ISSUE THREAD

**Issue by psifidotos** [psifidotos/nowdock-plasmoid#58]
What is coming up...

1. In **Plasma 5.9**, **transparent panels** without the need for a special theme, the user will be able to **keep its background constrast and blur effects**. (this has already been merged in plasma master by me ;) )
2. **huge re-write** for the Now Dock Panel. In cooperation with the plasma developers there were aspects of the implementation that couldnt get into the plasma main source code. Meaning that I will have to hack a lot :). But I believe very good things are going to come out soon ;),
meaning: 

- **dock masking**(the dock will give the unneeded space back to windows below it), the transparent areas are **real** transparent
- no need for the user to set the panel's **height and width**, it will be adjusted **automatically**
- **"Special Auto-Hidden"** State that updates tasks states and animations even when the dock is hidden
- **"Below Active - Fullscreen"** State (like the plank one, the dock goes below if an active of fullscreen windows overlaps with it)
- **"Always Visible"** State but with the windows to be able to go below the dock

**Comment by kupiqu** [comment]
It sounds great!

**Comment by psifidotos** [comment]
well my friends... :) 

almost everything is ready, and it works great!!! :)
I would like a lot to help me find issues before the **0.5** release... You can use **master branches** for the plasmoid and the panel...

**Importat Tips:**
1. Now Dock Panel from 0.5 version will need be compiled and installed through the **install-global.sh** script (**remove any local Now Dock Panel's installations before**). This is done because it is accompanied with a new c++ plugin which is doing all this great magic you are going to see ... :)
2. Now Dock Panel in **master** branch appears a green rectangle around its contents and various states. This is only for debugging purposes it wont exist in the final release, if you dont want it you can change in the main.qml file:
`at line 43:      property bool debugMode: false`

**Features**
Now Dock Panel supports 6 panel visibility states... (**Below Active, Below Maximized, Windows cover, Windows go below, Auto hide, Always Visible**), I havent implemented yet the Auto hide one but this is going to be a matter of days... All the other support beautiful sliding animations when then dock decides that should be shown or hide... :)

All these states work flawlessly with Plasma's Panel visibillity (Window Cover) which is also the default for all the new Now Dock Panels. For the **Always Visible** state you must set both the plasma's panel to "Always Visible" and the Now Dock's one. The height of the plasma panel in "Always Visible" state determines the snapping for windows etc.. So you could always have in "Always Visible" state your Now Dock appearing and take into account the snapping needed also...

**Comment by psifidotos** [comment]
**Auto Hide** has been just implemented...

**Comment by s8321414** [comment]
So nowdock plasmoid and panel will be released along with Plasma 5.9?

**Comment by psifidotos** [comment]
> So nowdock plasmoid and panel will be released along with Plasma 5.9?

no, they will probably be released before the mid of December, I will give 1-2 weeks for testing from today.
NowDock 0.5 is compatible with Plasma 5.8, it just needs the transparent-panel script in order to provide the plasma transparent panel. With Plasma 5.9 this will not be necessary, tranparent panels are supported inside Plasma.

a small change in the Now Dock Panel codepage is that it provide a c++ plugin which is doing its magic. So the user must download it from the store, build it and install it. This is why I am already preparing packages for openSuse. I hope that the various packages will include Now Dock 0.5 somehow in their repos in order to make the plasma users life easier...

**Comment by alexjp** [comment]
Hi .... sorry for the disappearance.

Trying out new 0.5 panel, it seems it doesnt respect the panel size: It always uses all bottom horizontal space.

**Comment by psifidotos** [comment]
> Trying out new 0.5 panel, it seems it doesnt respect the panel size: It always uses all bottom horizontal space.

is this needed? what is the case?

**Comment by alexjp** [comment]
> is this needed? what is the case?

well ... mac osX panel is not full width :)

seriously though, I prefer to not have it full width as I have edge corner mouse actions.

**Comment by psifidotos** [comment]
> seriously though, I prefer to not have it full width as I have edge corner mouse actions.

what are the edge corner mouse actions?

I use Present Windows corner and Show Desktop corner but they do not conflict with Now Dock, what is your case?

**Comment by alexjp** [comment]
well, I have on bottom corner ( where i have the panel ) for present windows and show applications menu also.

I just like non full width panels when i use them in autohide mode. simply a matter of taste

**Comment by psifidotos** [comment]
> well, I have on bottom corner ( where i have the panel ) for present windows and show applications menu also.
>
> I just like non full width panels when i use them in autohide mode. simply a matter of taste

this is fixed in master.... For Double Layout only Centered alignment based on MaximumLength is supported... For all the layouts maximumLength is taken into consideration ...

**Comment by alexjp** [comment]
> this is fixed in master.... For Double Layout only Centered alignment based on MaximumLength is supported... For all the layouts maximumLength is taken into consideration ...

Works wonders. The resize animation of the panel is very nice. :P thanks !

**Comment by psifidotos** [comment]
Well, I think that everything goes smoothly with the latest **master** branches, I havent found any **show stoppers bugs**, I will probably give a week for testers etc.

Packages for openSuse can be found at: [OpenSuse Build Service packages](https://software.opensuse.org/search?utf8=%E2%9C%93&q=now+dock&search_devel=false&search_unsupported=false&baseproject=openSUSE%3ALeap%3A42.2)

**Comment by alexjp** [comment]
Well I am on Gentoo, installed plasma git version.

It is working very smooth !

Current settings: Plasma panel autohide, nowdock panel "behind windows", central alignment and 8px background panel. Lovelly

**Comment by psifidotos** [comment]
@alexjp , nice to hear!!! :-)

**Comment by psifidotos** [comment]
guys, one confirmation please... Can you please try to add a new Now Dock Panel after installing the **master** branches ? https://github.com/psifidotos/nowdock-panel/issues/37   and send also what combination of plasma and frameworks are you using along with the results?
e.g. Now Dock can be added - Plasma 5.8.4 - Frameworks 5.28

**Comment by alexjp** [comment]
Well , for me, like sad earlier, everything is mostly working fine.

Some weird things that i think its not nowdock's fault: sometimes the dock doesn't go below a chromium window ( stays on top of chromium, although it is focused and using the space ).

other than that, its working how i like it :) i will report minor bugs that i think its nowdock's fault.

**Comment by psifidotos** [comment]
> Some weird things that i think its not nowdock's fault: sometimes the dock doesn't go below a chromium window ( stays on top of chromium, although it is focused and using the space ).

@alexjp , it could also be the Plasma panel setting "auto-hide" but there are plasma limitations that can not be overcomed from Now Dock. Have in mind that Now Dock support one main plasma panel setting "Windows can cover"...

**Comment by psifidotos** [comment]
> Some weird things that i think its not nowdock's fault: sometimes the dock doesn't go below a chromium window ( stays on top of chromium, although it is focused and using the space ).

@alexjp 
1. can you please test latest master branch for this, I found some situations that didnt hide the dock even they should, these situations are fixed in **master** with the latest commit...
2. can you please publish also your distro, plasma version and Frameworks version? There are some reports of problems to install the nowdock and I am trying to find a pattern why some people can not install correctly the Now Dock Panel...

Thanks a lot again!!!!

**Comment by alexjp** [comment]
1. I think it is better.... Now I am not able to catch it above chromium again.

2. My distro is Gentoo, I am using live ebuilds, meaning, compiling from git plasma master repositories.

**Comment by psifidotos** [comment]
> I think it is better.... Now I am not able to catch it above chromium again.

nice to hear...

> My distro is Gentoo, I am using live ebuilds, meaning, compiling from git plasma master repositories.

meaning you can drop the plasma transparent script, you can use any theme directly and you can keep your Blur and Background Constrast effects ;)

**Comment by psifidotos** [comment]
**I am thinking to add the plasmoid branch code inside the now dock panel branch code..**

Pros:
- I can use the Now Dock plugin inside the plasmoid without wondering if the plugin is installed by the user
- the user wont have to install two things but one
- everything will be in one place concerning forums, issues etc.
- the plasmoid and the panel will still have two separate entities for plasma
- We could rename the bundle in (Now Dock Suite)

What do you think?

**Comment by AndydeCleyre** [comment]
Your project is beautiful. I like the unification idea.

**Comment by alexjp** [comment]
Well ... if it makes sense in a technical code level ... why not ?

If it doesn't make sense.... I guess the user installing two things is not a problema, since if the Project takes off and lots of users use it, distro will make packages for it ( probably include it also by default )

**Comment by s8321414** [comment]
I think merge them into one is good :smiley:

**Comment by psifidotos** [comment]
I agree!!! Let's merge them... I will move the plasmoid codepage inside the panel one... The user will have to install only one thing, **not two** and everything should be the same after that... The user will have panel and plasmoid separately in its environment...

I will give a strict **NOTICE** in order to use only the Now Dock Panel codepage when that happens....

**Comment by kupiqu** [comment]
+1

**Comment by psifidotos** [comment]
**STRICT NOTICE** :)

the plasmoid has moved into the now dock panel branch... I took special care for the translation I hope that something didnt broke in the process!!! :) For me in greek works great... everything is ok...

the now-dock-plasmoid repo is not going to receive any updates here... All the changes will be added in the now-dock-panel...

if you install this new architecture you should see in the configuration windows for both the plasmoid and the panel the version 0.4.91

**Comment by psifidotos** [comment]
**I would like a lot your feedback around this: https://github.com/psifidotos/nowdock-panel/issues/49**   !!!!
